[MDB IGNORE] Angled Lights & Lighting Prototyping Tool

LemonInTheDark opened the pull request:
## About The Pull Request

Hello friends, I've been on a bit of a lighting kick recently, and I decided I clearly do not have enough things to work on as it is.
This pr adds angle support to static lights, and a concepting/debug tool for playing with lights on a map.

Let's start from first principles yeah?

### Why Angled Lights?

Mappers, since they can't actually see a light's effect in editor, tend to go off gut.
That gut is based more off what "makes sense" then how things actually work
This means they'll overplace light sources, and also they tend to treat lights, particularly light "bars" (the bigger ones) as directional.
So you'll have two lights on either sides of a pillar, lights inside a room with lights outside pointing out, etc.

![image](https://user-images.githubusercontent.com/58055496/228785032-63b86120-ea4c-4e52-b4e8-40a4b61e5bbc.png)

This has annoying side effects. A lot of our map is overlit, to the point that knocking out a light does.... pretty much nothing.
I find this sad, and would like to work to prevent it. I think dark and dim, while it does not suit the normal game, is amazing for vibes, and I want it to be easier to see that.

Angled lights bring how lights work more in line with how mappers expect lights work, and avoids bleedover into rooms that shouldn't be bled into, working towards that goal of mine.

### How Angled Lights?

This is more complex then you'd first think so we'll go step by step

![image](https://user-images.githubusercontent.com/58055496/228786117-d937b408-9bc2-4066-9aee-aae21b047151.png)

Oh before we start, some catchup from the last time I touched lighting code.
Instead of doing a lighting falloff calculation for each lighting corner (a block that represents the resolution of our lights) in view we instead generate cached lightsheets. These precalculate and store all possible falloffs for x and y distances from a source.

This is very useful for angle work, since it makes it almost totally free.
 
Atoms get 2 new values. light_angle and light_dir
Light angle is the angle the light uses, and light_dir is a cardinal direction it displays in

We take these values, and inside sheetbuilding do some optional angle work. getting the center angle, the angle of a pair of coords, and then the delta between them.
This is then multiplied against the standard falloff formula, and job done.

We do need some extra fenangling to make this all work nicely tho.

We currently use a pixel turf var stored on the light source to do distance calculations. 
This is the turf we pretend the light source is on for visuals, most often used to make wall lights work nice.
The trouble is it's not very granular, and doesn't always have the effect you might want.

So, instead of generating and storing a pixel turf to do our distance calculations against, we store x and y offset variables. 
We use them to expand our working range and sheet size to ensure things visually make sense, and then offset any positions by them.

I've added a way for sources to have opinions on their offsets too, and am using them for wall lights.
This ensures the angle calculations don't make the wall behind a light fulldark, which would be silly.

### Debug Tool?

In the interest of helping with that core problem, lights being complex to display, I've added a prototyping tool to the game.
It's locked behind mapping verbs, and works about like this.

Once the verb is activated, it iterates over all the sources in the world (except turfs because those are kinda silly), outlining and "freezing" them, preventing any future changes.
Then, it adds 3 buttons to the owners of a light source.
![image](https://user-images.githubusercontent.com/58055496/228776539-4b1d82af-1244-4ed6-8754-7f07e3e47cda.png)
The first button toggles the light on and off, as desired.
The third allows you to move the source around, with a little targeting icon replacing your mouse
The second tho, that's more interesting.

The second button opens a debug menu for that light
![image](https://user-images.githubusercontent.com/58055496/228777811-ae620588-f08a-4b50-93a0-beea593aea77.png)
There's a lot here, let's go through it.

Bit on the left is a list of templates, which allow you to sample existing light types (No I have no idea why the background is fullwhite, need to work on that pre merge)
You can choose one by clicking it, and hitting the upload button.

This replaces your existing lighting values with the template's, alongside replacing its icon and icon state so it looks right.
There are three types as of now, mostly for categorization. Bar, which are the larger typically stronger lights, Bulb, which are well, bulbs, and Misc which could be expanded, but currently just contains floor lights.

Alongside that you can manually edit the power, range, color and angle of the focused light.
I also have support for changing the direction of the light source, since anything that uses directional lighting would also tie light dir to it.
This isn't *always* done tho, so I should maybe find a way to edit light dir too.

My hope is this tool will allow for better concepting of a room's lights, and easier changing of individual object's light values to suit the right visuals.

### Lemon No Why What

Ok so I applied angle lights to bars and bulbs, which means I am changing the lighting of pretty much every map in the codebase.
I'm gonna uh, go check my work.

Alongside this I intend to give lighting some depth. So if there's room to make a space warmer, or highlight light colors from other sources, I will do that.

(Images as examples)
![image](https://user-images.githubusercontent.com/58055496/228786801-111b6493-c040-4199-ab99-ac1c914d034c.png)

I also want to work on that other goal of mine, making breaking lights matter. So I'll be doing what I can to ensure you only need to break one light to make a meaningful change in the scene.

This is semi complicated by one light source not ever actually reaching fullbright on its own, but we do what we must because we can.

![image](https://user-images.githubusercontent.com/58055496/228786483-b7ad6ecd-874f-4d90-b5ca-6ef78cb70d2b.png)

I'm as I hope you know biased towards darker spaces, I think contrast has vibes.
In particular I do not think strong lights really suit maintenance. 

Most of what is used there are bulbs, so I'm planning on replacing most uses with low power bulbs, to keep light impacts to rooms, alongside reducing the amount of lights placed in the main tunnels 

![image](https://user-images.githubusercontent.com/58055496/228786594-c6d7610c-611e-478b-bcba-173ebf4c4b12.png)

**If you take issue with this methodology please do so NOW**, I don't want to have to do another pass over things.
Oh also I'm saving station maps for last since ruins are less likely to get touched in mapping march and all.

### Misc + Finishing Thoughts

Light templates support mirroring vars off typepaths using a subtype, which means all the templates added here do not require updating if the source type changes somehow. I'd like to expand the template list at some point, perhaps in future.

I've opened this as a draft to make my intentions to make my changes to lights known, and to serve as motivation for all the map changes I need to do.

### Farish Future

I'm unhappy with how we currently configure lights. I would like a system that more directly matches the idea of drawing falloff curves, along with allowing for different falloffs for different colors, alongside extending the idea to angle falloff.
This would make out of engine lighting easier, allow for nicer looking lights (red to pink, blue to purple, etc), and improve accessibility by artists. 

This is slightly far off, because I have other obligations and it's kinda complicated, but I'd like to mention it cause it's one of my many pipedreams.

## Changelog
:cl:
add: Added angle lighting, applies it to most wall lights!
add: Adds a lighting prototyping tool, mappers go try it out (it's locked behind the mapping verb)
/:cl:

LemonInTheDark committed:
Adds a ui that allows for debugging lighting objects

So turning them off/on, changing their color, changing them to some
preset, changing their intensity/range, moving them around

This is intended to be used by mappers, testing lighting placements on
the map and seeing how many/of what sort they actually need

LemonInTheDark committed:
Adds angled lights, improves debug menu

Adds support for angled light sources
This is more complex then you'd first think so we'll go step by step

Atoms get 2 new values. light_angle and light_dir
Light angle is the angle the light uses, and light_dir is a cardinal
direction it displays in

We take these values, and inside sheetbuilding do some optional angle
work. getting the center angle, the angle of a pair of coords, and then
the delta between them.

This is then multiplied against the standard falloff formula, and job
done.

We do some extra fenangling to make this all work nicely tho

Instead of generating and storing a pixel turf to do our distance
calculations against, we store x and y offset variables. We use them to
expand our working range and sheet size to ensure things visually make
sense, and then offset any positions by them.

I've added a way for sources to have opinions on their offsets too,
and am using them for wall light

This ensures the angle calculations don't make the wall behind a wall
light fulldark, which would be silly.

Debug things

I've added support for changing the direction of atoms in the debug
menu, which is used by most lights to set their direction

Added support for freezing an atom's light, so templates alone can
control it

Oh right I never described those

Adds lighting templates. Global singletons that describe a configuration
of lighting, matched against an icon state and name for use in the ui

The ui allows for scrolling between them, and they support mirroring
default vars off typepaths, to ensure parity through changes

They don't support light offsets rn, maybe someday

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into light-debug

LemonInTheDark committed:
Does some housekeeping, renames a typepath and makes the ui nicer

RikuTheKiller commented:
Good, was getting tired of chasing 20 billion lights to cause even a minor overall effect.

LemonInTheDark committed:
damn "other people"

LemonInTheDark committed:
yeeeet

LemonInTheDark commented:
I could pr this to a project branch if you'd like, unsure if that's wanted

lessthnthree commented:
Great! The dimmer environment allowing lights to actually create ambiance is something we did downstream and it looks better. It'd be nice to have it in the base code.

CRITAWAKETS commented:
might be good to have angled lights also have a small regular light radius (pretty much how lights _currently work_) but dimmer, that way it kinda mimics how light would also bounce around. it could also help if certain wall shapes end up looking weird

LemonInTheDark commented:
> Unrestricted doors emit light (as we've spoken about), should we consider that a light source or were you planning to do something else about that
> 
> Im okay either way just noticed that doors didnt have the config buttons

They aren't like, real light sources. They're overlays sitting on the above lighting plane.
This was done intentionally because of yaknow nightmares.

Frankly the implementation is repulsive to me, and not just cause I don't like directional all access doors in maint.
It looks horrible, especially when walking around, doesn't interact with the world, and just like GAH.
The trouble is a better visual cue would require a sprite modification, and that's harder so it's not been done. Makes me sad.

> Great! The dimmer environment allowing lights to actually create ambiance is something we did downstream and it looks better. It'd be nice to have it in the base code.

Darker spaces have a lot of appeal yeah. I don't like how far skyrat has gone with it, and I don't plan on going anywhere near that. I think fullishbright is good for rooms with no contrasting light sources, since it makes sure sprites are visible.
That said giving players control over the dark is I think good, and making rooms that aren't "primary" dimmer has upside.

> might be good to have angled lights also have a small regular light radius (pretty much how lights _currently work_) but dimmer, that way it kinda mimics how light would also bounce around. it could also help if certain wall shapes end up looking weird

Potentially. It'd require changing how I do the angle math to sort of interlace with distance. I'm using * against the distance here because it looks better to me then the prototyping I've done with the other method.
The other option is taking an angle output of the function and feeding it into the distance calc. This works, but it's harder to get a nice gradient. 
It's harder to poke at this stuff then I'd like, which is something I'd like to change someday. We'll see.

LemonInTheDark committed:
Fixes lighting runtimes from queues changing in init during a sleep

san7890 commented:
> They aren't like, real light sources. They're overlays sitting on the above lighting plane. This was done intentionally because of yaknow nightmares.
> 
> Frankly the implementation is repulsive to me, and not just cause I don't like directional all access doors in maint. It looks horrible, especially when walking around, doesn't interact with the world, and just like GAH. The trouble is a better visual cue would require a sprite modification, and that's harder so it's not been done. Makes me sad.

That's fair. Imo this would be the biggest obstacle to truly getting good looking maintenance at the level you want it given how many doors we can have up to so you might want to see about chasing down this better implementation in another PR so this one can stick the landing as best as possible.

I know @zxaber keeps postulating ideas on making the unres helpers look better, but I forget what their ideas were.

LemonInTheDark commented:
I believe the holdup for that was wallening for one reason or another

LemonInTheDark committed:
CLOWN BIODOME: Moves some lights around, makes some corners properly warm, tries to keep things roughly bright

LemonInTheDark committed:
SNOW BIODOME: Moves some lights around in the trophy room to give off more blue. The space is a bit darker then before but it had dark areas previously, so I think it's mostly just vibes

LemonInTheDark committed:
GREED RUIN: Moves lights around to emphasize the center gold pile, and make things feel a bit more lonely

LemonInTheDark committed:
SEED VAULT: Removes a LOOOT of redundant lights. Things should still be about as bright, just uh, more maluable

LemonInTheDark committed:
SURVIVAL POD: Changes a small light to a broken small light, for vibes. The storage has a nice green glow that I think fits this space

LemonInTheDark committed:
SYNDICATE BASE: Ensures brightness, highlights floor lights, retains some nice darkness

LemonInTheDark committed:
GOLEM SHIP: less lights in the main room, shuffles some things around to light things better

LemonInTheDark committed:
ASTEROID SHUTTLE: Floor lights to avoid directional memes, some redirecting to keep lighting contained

LemonInTheDark committed:
VENTURE SHUTTLE: realigned lights, cut down on redundants, added a cold light for the meeting space

LemonInTheDark committed:
BAR SHUTTLE: highlights warm lights near the doors better

LemonInTheDark committed:
PODS SHUTTLE: Uses just warm lights to make things feel a bit more lonely

LemonInTheDark committed:
USSR SHUTTLE: Much less fullbright, moves some lights around to highlight medbay better while still vibing, red lights in the pit cause yaknow, and a dim red light in the bear den so opening the doors has punch

LemonInTheDark committed:
LUXARY SHUTTLE: Moved a plant, made the shit side darker, removed bleedin from the nice side. Little bit of other moves but nothing that major

LemonInTheDark committed:
MEDIEVAL SHUTTLE: Reduced lights in cockpit and main room to emphasize spacelight, reduces light near entrances to pull people into the center, reangled lights to prevent some bleed and properly light things, replaced outside small bulbs with floor lights, they do the job better

LemonInTheDark committed:
DISCO INFERNO SHUTTLE: Almost no notes, just a wall light sitting on a window -> floor light, for vibes and to prempt future changes

LemonInTheDark committed:
BIRDBOAT SHUTTLE: removed a light, flipped a few others to cut down on bleed

LemonInTheDark committed:
BOX SHUTTLE: Angled lights, warmer lower left, removed some redundant lights

LemonInTheDark committed:
DELTA SHUTTLE: moves some lights around to cover for darkness/reduce duplicate brightness sources, flips some things around to prevent bleeds. Most of the thing is well lit, sec containment isn't but I think it goes pretty well with the floor tiles. We rely on space to light up the very edges of the doors

LemonInTheDark committed:
CLOWN SHUTTLE: Uses bulbs to create a warmer tone, going for sorta semi sleepover vibes. I'm a bit unsure on this one could use a vibecheck

LemonInTheDark committed:
META SHUTTLE: Removes redundant lights, preventing some overflow. Reduces light in cockpit slightly, using a desk lamp instead of a second bar

LemonInTheDark committed:
KILO SHUTTLE: Bit dimmer, use of floor lights, using warmer lights for the two side compartments since I think it looks nice. replaces 2 bars with 2 bulbs and a floor light in the bridge, I think it looks nicer

LemonInTheDark committed:
MINI STATION: Floor lights in the bridge, moves lights on shuttle around a bit. The brig and medbay get a bit dimmer, but I don't think it's THAT bad. Unsure

LemonInTheDark committed:
TRAM SHUTTLE: Removes intercompartment small lights, they're redundant because of space. Moves some stuff around in the second compartment to account for angles, swaps bars for bulbs and a floor light in the bridge, something something fullbright

LemonInTheDark committed:
SCRAPHEAP SHUTTLE: Rotates a light to ensure suicide door coverage, moves a bulb in the brig around to make it work proper, places another bulb by the recharger to cover newly dark space. Replaces bar in bridge with a desk light. It works pretty damn well and I think fits the vibe really well

LemonInTheDark committed:
PUBBY SHUTTLE: Most compartments have 2x more bars then needed, rectifies that, uses bulbs for the prisoners for vibes

LemonInTheDark committed:
CERE SHUTTLE: Removes some unneded lights, replaces some with bulbs near enterences and the bathroom. Moves the mechroom around a little to focus the lights in the area of interest. Replaced two bars in the bridge with bulbs for better contrast and to avoid overlighting the space

LemonInTheDark committed:
SUPERMATTER SHUTTLE: Makes the bottom two bars warm, top two into bulbs and shifts em up. Trying to provide contrast against the green lasers cause they look quite striking

LemonInTheDark committed:
DANIEL SHUTTLE: removes two lights. We get almost the same coverage outside some darkness at the edges which I'm ok with, and it makes the vandalism more impactful

LemonInTheDark committed:
GOON SHUTTLE: Commits high treason by replacing 2 lights with one

LemonInTheDark committed:
BBALL SHUTTLE: Reduces wall light count signficantly, focusing lights on the court using floor lights. Swaps wall light in tunnel with bulb to allow the red bulb in the brig to actually light, removes second hallway light that existed BECAUSE OF TINTED WINDOWS (work assumes tinted windows get nuked), moves medbay lights up a tile to better cover the space. Adds a bulb to the top left to light the crate and the door

LemonInTheDark committed:
WABBAJACK SHUTTLE: New lights in the 2 aux rooms to make sure they're nicely lit. I'd like to do more to the main room but I can't really, and the bulbs being auxilary is... ok

LemonInTheDark committed:
OMEGA SHUTTLE: Adds a wall to the sec door to match the command side, moves command and sec lights down to the wall to avoid bleed. Removes bottom bar light, brings the 2 bulbs down, moves bars in 2 stow areas down too. This provides full lighting without overlighting the space

LemonInTheDark committed:
CRUISE SHUTTLE: Holy shit many there's so many duped lights. Half the ones in the hallway are basically irrelevant (people aren't staying there and it's good for it to go fulldark with one or two breaks). By the disco floor the disco lights were fully blown out, so I swapped some lights for bulbs. (similar at the dating tables). In the holo room, almost all the lights were irrelevant, since those plants emit light. In the first shuttle area most of the bar lights were irrelevant, so I swapped em for bulbs to ensure fullishbright. The second shuttle area is covered in glow plants, so we only really need two lights. I moved the lights in the cockpit around to make yeeting one a bit more impactful, and to provide space for starlight

LemonInTheDark committed:
NATURE SHUTTLE: Rotates inner floor lights 45 degrees, looks better this way. Removes half the ring lights, they were just redundant. Removes most of the lights in the bridge, moves the one on the bottom to the far left so it can provide a bit of fullbright to the room on the left, which I'm leaving with just one bar. the rule of thumb is you need 2 bars to be bright enough on dark tiles, but the light leaking in serves this purpose well enough to work

LemonInTheDark committed:
CASINO SHUTTLE: Jesus christ man. OK half these lights are redundant, removes the bottom center, replaces it with some floor lights and just the angled lights from the brig (why are there so many lights in the brig, angles the lights in medbay and cargo to prevent drift, adds warm lights focused on the beds in the bridge sideroom, makes the bridge into more of a lounge with dimmer lights

LemonInTheDark committed:
SHADOW SHUTTLE: Plasma chamber and hallways have more then they need, the issues in the center can be solved by two floor lights, and the two chambers, pulling lights from outside, really only need one floorlight slightly offset. I removed two bars from the bridge and replaced them with bulbs. Oh and defaults the disco ball to on, for the meme

LemonInTheDark committed:
FISH SHUTTLE: Way too many lights, made medbay semi dim, chasm room dim warm, maint room warm, used floorlights in hallway, floorlights and 2 bars in sec instead of 4 bars, 2 bulbs and a floor tile in the bridge instead of like 4 bars or something insane

LemonInTheDark committed:
LANCE SHUTTLE: Floor lights in a few rooms instead of beacons and bars, reduced lower room from 4 bars to 2 bars and 2 bulbs, moved a bar in the bridge for better coverage. The main crew space is getting dimmer here from angles, I kinda like it tho so I left it

LemonInTheDark committed:
TRANQULITY SHUTTLE: Reduces wall light count, makes backroom warmer, removes a few floor lights that blew out starlight and a desk lamp, moves a floor light a tile to the right (kinda breaks a visual lineup, would like feedback), dims medback and sec cause it works alright like this, makes the bridge more vibes with floor lights instead of 4 bars causing fullbright

LemonInTheDark committed:
DELTA SHUTTLE: Removes some unneeded lights from main area, moves a light in the gear room to light the ssus, and moves a light in the command main hall to avoid blowing out the command area

LemonInTheDark committed:
RAVEN SHUTTLE: Almost no notes. Removes a bar that overlit the bridge, replaces it with a floor light. swaps 2 more bars for floor lights in the 2 midpassages to engi, and adds a floor light to the bridge main entrance

LemonInTheDark committed:
ZETA SHUTTLE: Removes bars sitting directly on top of glowing plants. Realigngs lighting to better avoid crossover. Uses floor lights in medbay, and the back hallway. Removes bars in the 2 back rooms that don't do much, replaces bars with floor lights in brig to allow space to the plants

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into light-debug

LemonInTheDark commented:
alright. that's all the escape shuttles, and all the lavaland ruins.
I am so tired

LemonInTheDark committed:
makes lighting init properly, fucked that one up huh

LemonInTheDark committed:
avoids readding the debugger if it's already there

LemonInTheDark committed:
go home

LemonInTheDark committed:
RESEARCH RUIN: removes some broken lights that'd make the space fullbright otherwise, breaks a light to focus on starlight, swaps a bar and a bulb to avoid fullbright in a room

LemonInTheDark committed:
OLDSTATION RUIN: removes a light in engi, moves a light in mining to account for angles. Removes a light in the corridors, breaks a few others to lead to jumpscares and vibes. Breaks a light in viro to shadow the fallen chair, removes dupe lists in the research room, removes one in viro, breaks some lights in the nest for tension. Adds a crate of light bulbs/tubes to be nice

LemonInTheDark committed:
ANOMALY RUIN: less over bright, particularly in the broken lights area

LemonInTheDark committed:
ATMOS ASTEROID: bulbs in the gas containers, moves some things around to have a bit of a more darkness prone space

LemonInTheDark committed:
AUTHOR RUIN: WAY too many lights, uses floor lights , 2 bulbs and a single bar in the writing room to light the space. Adds a desk lamp to the lower corner, and a light switch for sleeping

LemonInTheDark committed:
ZOO RUIN: Way too many lights, swapped window bars for floor lights, added a broken floor light varient for the xeno room (which I also made dark cause the moss glows)

LemonInTheDark committed:
CLERIC'S DEN: Makes the placed light prisms actually... light. Come on man\

LemonInTheDark committed:
CYBORG MOTHERSHIP: Removes beacons since they just white out the darkness. Replaces the 4 bars with 1 at the base (this room would be well lit by its floors if it had power), replaces 2 bars in top room with 1 broken and 1 non broken floor light, was too much

LemonInTheDark committed:
DERELICT SULACO: Reduces light count on bridge using floor lights, moves lights around in hold to allow space for the moss, and to not in theory overlight the space. Turns on a desk lamp and uses it to replace a bar, replaces some other bars with floor lights

LemonInTheDark committed:
BIG DERELICT: Fixes mapped glowshrooms self nuking after a short period, Moves some lights around to bring things from fullbright to half dim

LemonInTheDark committed:
Adds toggle support to light debugging

LemonInTheDark committed:
DJ STATION: Less lights, bulbs > bars most of the time. It's hard to do this properly because IT'S SPLIT 4 WAYS AND THE LIGHT BLEEDS FUCK YOU THUNDER but it's good enough

LemonInTheDark committed:
DK RUIN: Removes some redundant lights, dims the tube room

LemonInTheDark committed:
MROW EXPERIMENT RUIN: Replaces a bar with a bulb, shifts one down at ile

LemonInTheDark committed:
HELL FACTORY: Dims hanger slightly, removes a light by the door in maint for yaknow reasons

LemonInTheDark committed:
HILBERT RUIN: Reduces light count. floor lights replaced with bars and glass in the hydro bay, focus on lamps in small rooms

LemonInTheDark committed:
DRONE RUIN: Redoes some areas to prevent improperly powered lights

LemonInTheDark committed:
MASS TRANSIT RUIN: 4 lights become 2, works just fine

LemonInTheDark committed:
SMOOTHIE RUIN: reduces light count by 2, retains levels well

LemonInTheDark committed:
DEEP STORAGE: less lights, more dim spaces fitting the bunker vibe, more room for the light room to breath

LemonInTheDark committed:
FORGOTTEN SHIP RUIN: Removes some lights, replaces some with bulbs for contrast

LemonInTheDark committed:
LISTENING STATION: Tints the beacons blue, cause white looks fucking bad. Rejiggers the main room to be a bit less eye posion while retaining the fucked frathouse vibe. That red bar light wasn't doing favors, and I'd like the red light over the bomb to show more. Also replaces red bar with bulb.

LemonInTheDark committed:
FACEOFF RUIN: Reduces light count, flips some around so they light properly

LemonInTheDark committed:
LIZARD'S GAS: Dims shop, removes a light replaces another with a bulb. Uses the power of brightness to drive snack sales. Removes a few lights around the sign (I really dislike beacons now)

LemonInTheDark committed:
SPACE HOTEL: Dims things a bit, provides color space for the candles in the date room and the floor on the admin area. This space can stand dimmer lights and look nice because of the floor tiles, and the power of highlighting with light is nice, so I'm gonna stretch it. May have gone a bit far in the right lounge area, unsure

LemonInTheDark committed:
TURRETED OUTPOST: dimmer secondary rooms, I'm using bleed here to light the computer table, and grouping lights in the armory to focus on the good shit. Using bulbs in the sleep room cause vibes

LemonInTheDark committed:
WAYSTATION RUIN: Almost no notes. flips a bulb in the bathroom to work, removes a light that didn't need to be there (specially if the place should feel trashed)

LemonInTheDark commented:
Knocked out all the space ruins, sans Traveler's Rest, since I need to figure out how to handle that with the mapper.
Made mapped in glowshrooms not decay because fucking christ, and added a broken variant for floor lights

LemonInTheDark committed:
for the mdb nerds

LemonInTheDark committed:
BUG HABITAT: Removes a light, moves another down so it covers the space better

LemonInTheDark committed:
MOTH PIZZA: Moves some stuff around to make the actual running appearance a bit dimmer in the main room, I think I prefer it like this. Doesn't impact typical visuals all that much tho

LemonInTheDark committed:
ICEHERMIT: THIS RUIN HAD 5 LIGHTBARS AND NEEDED 0 OF THEM. The fire and the fairy grass do a perfectly fine job lighting the space

LemonInTheDark committed:
ICEMAILROOM: Removes a light to give entrance room more vibes

LemonInTheDark commented:
Alright, there's all of icebox

LemonInTheDark committed:
BIRDBOAT CARGO: removes a light, overlit

LemonInTheDark committed:
PUBBY CARGO: Removes 2 bars, adds a bulb to the lower left. Moves the one up top to the upper left. This has a sort of dim pastoral effect, which I think matches

LemonInTheDark committed:
BOX CARGO: Removes a redundant light, does the top bottom trick to light the center well

LemonInTheDark committed:
DELTA CARGO: Pretty similar to pubby, 2 bulbs and a bar to give a warmer impression

LemonInTheDark committed:
whoops

LemonInTheDark committed:
FERRY PASS: removes lights from the kilo ferry, adds them to the code rotten lighthouse, basic and meat ferry

LemonInTheDark committed:
CERE WHITESHIP: Moves some lights around, reduces count in main hold from 4 to 2, bright in center dim at edge, swaps 2 lights in bridge for 2 bulbs and a floor lamp

LemonInTheDark committed:
DONUT WHITESHIP: Swaps 2 bars for 2 bulbs and a floor light, I like the asthetic better, and it's close to the same for this small a space

LemonInTheDark committed:
TRAM WHITESHIP: Removes effectively fullbright lightbars in the two lower cabins, replacing them with a bulb. The left cabin got a broken bar too, since there's a desklamp. Removes all the lights on the walkway, they only served to overlight the shuttles, and since the walkway self lights it's sorta pointless. Removes 2 lights from the lower airlock, and 1 from the upper, replacing the lower 2 with a floor light, and the upper one with a broken light to replace. ALSO, UNOPACITIES THE WINDOWS AND DOORS ON THE RIGHT POD. FUCK YOU PAXILMANIAC I WILL SHOWER IN YOUR BLOOD (I hate directional opacityihatedirectionalopacityihatedirectionalopacity)

LemonInTheDark committed:
PERSONAL WHITESHIP: Swaps a cold light for a broken one, better on vibe

LemonInTheDark committed:
OBELISK WHITESHIP: Swaps two bars for a floor light

Rex9001 commented:
Should have the tool label, no?

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into light-debug

LemonInTheDark committed:
MIMES VS CLOWNS RUIN: Removes a light, swaps a bar for a bulb for vibes

LemonInTheDark committed:
BOTANICAL HAVEN: Reorients a light to light the room proper, moves one up to highlight the tree

LemonInTheDark committed:
THE OUTLET: Few rooms are dimmer, the engi room. Breaks some lights for asthetics, moves a light in the disco room to more fully light the space. Adds an extra bulb to the research room to fully light it

LemonInTheDark committed:
INTERDYNE RUIN: Removes maybe half the lights. doesn't really impact most of the space, it's quite overlit

LemonInTheDark committed:
INFESTED FRIGATE RUIN: Removes some disabled lights, moves things around to light better both broken and fixed\

LemonInTheDark committed:
TRANSITY BOOTH RUIN: bar -> bulb, moves a bar to the back to better light the space. Makes things slightly dimmer, but more vibes

LemonInTheDark committed:
ARRIVALS BOX: 2 less lights, not sure how I feel about this. Moves an intercom to not fight with the light bars

LemonInTheDark committed:
ARRIVALS DONUT: 4 -> 2, still a bit unsure but more ok with this

LemonInTheDark committed:
ARRIVALS DELTA: 2 -> 3, angles are a bitch

LemonInTheDark committed:
ARRIVALS KILO: 4 -> 2, looks better cause she's a bit smaller

LemonInTheDark committed:
ARRIVALS PUBBY: 4 -> 2, similar to kilo this works quite well cause it's smaller

LemonInTheDark committed:
ARRIVALS NORTHSTAR: 4 -> 2, moves an intercomm

LemonInTheDark commented:
Alright that's arrival, cargo, ferry, whiteship, mining and labor shuttles. Halfway through the categories

LemonInTheDark committed:
INFILTRATIOR BASIC: A few bar -> bulb, some bar -> floor, a tad dimmer but I think to an acceptable level. Mostly mirrored, not totally cause I want to highlight the circuit floors in the cyborg room

LemonInTheDark committed:
INFILTRATOR ADVANCED: Few bars to bulbs, some bars to floor lights or just straight removals

LemonInTheDark committed:
PIRATE SIVERSCALE: Removes some lights from the bridge/bridge area, provibes space for the fireplace in future, reduces lights in hold from 3 to 1 cause it's the hold

LemonInTheDark committed:
PIRATE DUTCHMAN: Uses floor lights for the stripper poles, removes the ones on the catwalks (starlight), and fenangles for angles

LemonInTheDark committed:
PIRATE PSYKER: Less lights in the bedroom and laundrymat, less light in the trophy room, swaps a bar for a bulb in the bridge, removes a reudndant broken light in the hold. Swaps around a bulb in the engi room to ensure the sign is fully lit

LemonInTheDark committed:
HUNTER SHUTTLE (COP): Ok so this used to be fullbright, so I'm changing that. While we're here, lets do a bar and a bulb for the hold, and a floor light for the cabin

LemonInTheDark committed:
HUNTER SHUTTLE (BOUNTY): Used to be fullbright, some floor lights and bulbs to give a sort of sus but not fully funded vibe

LemonInTheDark committed:
HUNTER SHUTTLE (RUSSIAN): This one was actually meant to be lit (didn't work but yaknow). Mostly alright, I removed a few lights to add texture cause it was a bit much. Some reliance on door bleed to make sure things look alright

LemonInTheDark committed:
DROPSHIP RUIN: Bit of moving about for angles, nothin major

LemonInTheDark committed:
ERT SHUTTLE (BOUNTY): Dupe of the bounty hunter shuttle but sans bounty hunter stuff

LemonInTheDark commented:
and there's the rest of the shuttles (sans some away mission memes I'll need to look at in context)

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into light-debug

LemonInTheDark committed:
WIZARD SHUTTLE: moves some lights around to avoid fullbright memes and do angles better, s it

LemonInTheDark committed:
temp

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into light-debug

LemonInTheDark committed:
NINJA DEN: Moves some lights to work with angles, makes a few spaces a bit dimmer, more moodey. Unsure on this one we'll have to see

LemonInTheDark committed:
NUKIE BASE: Makes uranium statues glow green (wtf old coders). Less light bars, some bulbs used selectively, dims the docking port a lot, I want it feel sorta grungy. Nukes all the directional opacity, life is pain

LemonInTheDark committed:
Allows lighting atoms to swap dir on move, even if frozen

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into light-debug

LemonInTheDark committed:
LAVALAND: Bit dimmer all round, sorta dank, might be too much thinking of throwing this at people

optimumtact commented:
Do you need a map freeze to finish this?

optimumtact commented:
it's getting enough attention that im okay with it for now

LemonInTheDark commented:
*cracks nuts*

LemonInTheDark committed:
LAVALAND: Moves a few lights around, removes one in the robotics ward

LemonInTheDark committed:
DINER RUIN: Removes a lot of redundant lights (why are these all dim), makes the freezer room cold

LemonInTheDark committed:
DEEPSTORAGE RUIN: Removes one light in the side room, it was redundant

LemonInTheDark committed:
PRISON SHIP RUIN: Removes some doubled up broken bars, replaces some with bulbs. dims the left room by breaking its bulb (it's slightly lit by the right side when the door is open. Makes the lower bar /dim)

LemonInTheDark committed:
ignores a few ruins

LemonInTheDark committed:
BIRDSHOT ARRIVAL: Removes 2 bars, uses 2 cold bars and 2 floor lights to light the shuttle

LemonInTheDark committed:
BIRDSHOT CARGO: Removes 2 bars, making the space a bit less fullbright

LemonInTheDark committed:
BIRDSHOT EMERGENCY: Removes some bulbs that are out in space, removes 2 bulbs and reorients to acomplish a similar thing

LemonInTheDark committed:
NORTHSTAR EMERGENCY: Removes 2 bars on the top/bottom left, they were mostly redundant

LemonInTheDark committed:
STARFURY: Replaces some bars with bulbs/floor lights, moves stuff arounnd to work better, ensures weapons room doesn't bleed out the red

LemonInTheDark committed:
CAVES: Removes one redundant bar

LemonInTheDark committed:
MOON OUTPOST: Removes 2 bars in the grav gen room

LemonInTheDark committed:
RESEARCH: Removes some duped bars, move some around to not crowd wallhangings or bleed too bad

LemonInTheDark committed:
SNOW CABIN: Goes over the lights of the cabins, removes some fullbright. Replaces the area lighting with sources over the whole map, colored lightly blue (this is maybe a bit much, but this would only really hurt on load, so I don't mind... that much. Skylighting would be better tbh

LemonInTheDark committed:
SNOWDIN: Cleans up some weird areas and overlit spaces, this was decently small

LemonInTheDark commented:
Oh also still need to deal with that star shaped lad, not sure what to do about him as of yet

LemonInTheDark commented:
> Should have the tool label, no?

Our labels aren't really concrete, but iirc that applies to external tooling, rather then stuff like this. Can't really recall tho, don't mind too much

optimumtact commented:
>Gonna hold off on the station maps till I get confirmation that the map team is on board.

@tgstation/maptainers I SUMMON YOU

Maurukas commented:
This is extremely impressive and very interesting.

One of the problems with lighting at the moment is that the workflow requires you to make changes in the map editor, boot up a server, and look at them in game. If it doesn't work right, you must update the map, reload the server and look again. Repeat this process until it looks right. This means the standard for lighting is just to slap down more bulbs than you need

The ability to tweak lighting in-game, find the appropriate settings and values, then apply them in the map editor once is faster and less tedious. Any improvement to the mapping tools is appreciated.

LemonInTheDark commented:
Fuck

jlsnow301 commented on tgui/packages/tgui/interfaces/LightController.tsx:
```diff
@@ -0,0 +1,319 @@
+import { Box, Button, Knob, Section, Slider, Stack, Tabs } from '../components';
+import { Window } from '../layouts';
+
+const NORTH = 1;
+const SOUTH = 2;
+const EAST = 4;
+const WEST = 8;
```
these could be made into an enum

jlsnow301 commented on tgui/packages/tgui/interfaces/LightController.tsx:
```diff
@@ -0,0 +1,319 @@
+}
+
+type Data = {
+  on: boolean;
```
```suggestion
  on: BooleanLike;
```

you'll need to import this as well

jlsnow301 commented on tgui/packages/tgui/interfaces/LightController.tsx:
```diff
@@ -0,0 +1,319 @@
+                      key={category}
+                      selected={currentCategory === category}
+                      onClick={() => setCurrentCategory(category)}>
+                      <Box fontSize="14px" bold textColor={"#eee"}>
```
```suggestion
                      <Box fontSize="14px" bold textColor="#eee">
```

jlsnow301 commented on tgui/packages/tgui/interfaces/LightController.tsx:
```diff
@@ -0,0 +1,319 @@
+            color="olive"
+            minValue={-1}
+            maxValue={5}
+            format={(value) => { return round(value, 2); }}
```
do you need to wrap this in curly braces with a return here? should be able to just value => round(

LemonInTheDark replied:
yep, you're right

jlsnow301 commented on tgui/packages/tgui/interfaces/LightController.tsx:
```diff
@@ -0,0 +1,319 @@
+                icon="brush"
+                textColor={info.color}
+                onClick={() => act("change_color")}
+                content={info.color}
```
w/ buttons, please put the content as a child rather than as content
(if this causes textcolor to not work, nvm)

LemonInTheDark replied:
No I just thought this was preferred is all

jlsnow301 commented on tgui/packages/tgui/interfaces/LightController.tsx:
```diff
@@ -0,0 +1,319 @@
+
+export const LightController = (props, context) => {
+  const { act, data } = useBackend<Data>(context);
+  const { light_info, templates, default_id, default_category, category_ids } = data;
```
```suggestion
  const { light_info, templates = [], default_id, default_category, category_ids } = data;
```
safety first

jlsnow301 commented on tgui/packages/tgui/interfaces/LightController.tsx:
```diff
@@ -0,0 +1,319 @@
+          <Stack.Item>
+            <Section fitted fill scrollable width="170px">
+              <Tabs fluid centered>
+                {Object.keys(category_ids).map(
```
this should be made into a const earlier too just to be safe. something like
```ts
const categories = category_ids ? Object.keys(category_ids) : [];
```

LemonInTheDark replied:
o7

jlsnow301 reviewed (commented):
minor nits. good work

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into light-debug

LemonInTheDark committed:
Passes over syndi lavaland base, cere and birdshot whiteships. Mostly moving around lights, fixes an issue w turf passthrough in modular syndicate stuff. Also cleans up some code broken by the merge

LemonInTheDark committed:
you didn't see this

LemonInTheDark committed:
saves pubby

LemonInTheDark committed:
prettifys the tsx, addresses review comments

LemonInTheDark committed:
Adds a dim bulb subtype, removes ids from templates, starts in on DELTA STATION: Upper maint, removed some lights, dimmed some others, moved stuff around to make things darker/more moody

LemonInTheDark committed:
DELTA STATION: All of maint, darker, enhance, etc

LemonInTheDark committed:
DELTA STATION: Engineering, lots of duped up lights here. Potentially too far gonna run by someone later

LemonInTheDark committed:
Finishes engineering on delta, starts to extend into command

LemonInTheDark committed:
Command and sec, let's goooo

LemonInTheDark committed:
delta, ai sat. FUCKING VIBES I love colored floors

LemonInTheDark committed:
delta xenobio, research director's room/halls

Oh also I made fireplaces actually colored like fire, and gave them
directional lighting/dir offsets so they'd look nice"

LemonInTheDark committed:
rest of sci

LemonInTheDark committed:
medbay, departures, chapel

LemonInTheDark committed:
Dorms, arrivals, service (DONE WITH DELTA LET'S GOOOO)

LemonInTheDark commented:
Starting in on station maps, finished delta station. I'm mostly happy with it tho I may have been a bit too hard in some places.
I've asked the maptainers to hold off on major changes for about 2 weeks so I can get the rest of the stations knocked out. I don't want to get stuck in conflict hell.

LemonInTheDark committed:
METASTATION: Arrivals, Cargo, AI SAT, Command, Library, Sec. (It's so much easier when everything isn't gigantic fuck you delta. Mostly moving lights around, tried to mood spaces may have gone too heavy as always

LemonInTheDark committed:
META: Medbay and Sci. Good vibes I think, maybe a bit heavy on the blue

LemonInTheDark committed:
Gives floor lights a valid range, METASTATION: Covers service and engi. Still needs a quality pass but this is a good base

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into light-debug

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into light-debug

LemonInTheDark committed:
Fixes a few issues with debug toggling

LemonInTheDark committed:
ICEBOX: Sec, arrivals, service, ai sat

LemonInTheDark committed:
Rest of icebox

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into light-debug

LemonInTheDark committed:
Adds a new button that temp disables all lights impacting the same corners as our source

I've been doing this manually, hopefully it makes my life and others
lives easier in future

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into light-debug

LemonInTheDark committed:
New light ui, cleans up template swapping

Instead of swapping templates changing visuals, it spawns a new object
Hopefully this eases issues with angles and such

Adds a new ui/action button that lets you easily create lights as
desired

LemonInTheDark committed:
BIRDSHOT: security

LemonInTheDark commented on code/modules/lighting/lighting_corner.dm:
```diff
@@ -192,3 +192,40 @@
 
 	return ..()
+
+/datum/lighting_corner/proc/display(max_lum)
+	if(QDELETED(src))
+		return
+
+	var/turf/draw_to = master_SW || master_NE || master_SE || master_NW
+	var/mutable_appearance/display = mutable_appearance('icons/turf/debug.dmi', "corner_color", LIGHT_DEBUG_LAYER, draw_to, BALLOON_CHAT_PLANE)
+	if(x > draw_to.x)
+		display.pixel_x = 16
+	else
+		display.pixel_x = -16
+	if(y > draw_to.y)
+		display.pixel_y = 16
+	else
+		display.pixel_y = -16
+
+	var/index = largest_color_luminosity / max_lum
+	display.color = gradient(0, COLOR_BLACK, 1, COLOR_WHITE, index)
+
+	draw_to.add_overlay(display)
+
+/datum/lighting_corner/dummy/display()
+	return
+
+/proc/display_corners()
+	var/list/corners = list()
+	var/max_lum = 0
+	for(var/datum/lighting_corner/corner) // I am so sorry
+		corners += corner
+		max_lum = max(max_lum, corner.largest_color_luminosity)
+
+
+	for(var/datum/lighting_corner/corner as anything in corners)
+		corner.display(max_lum)
```
redundant, need to remove I think? It's a debug tool I was making to help visualize lighting corners, may keep it as a mapping proc just for the sake of it? unsure.

LemonInTheDark commented on code/modules/asset_cache/assets/light_templates.dm:
```diff
@@ -0,0 +1,7 @@
+/datum/asset/spritesheet/lights
+	name = "lights"
+
+/datum/asset/spritesheet/lights/create_spritesheets()
+	for(var/id in GLOB.light_types)
+		var/datum/light_template/template = GLOB.light_types[id]
+		Insert("light_fantastic_[template.id]", template.icon, template.icon_state)
```
Need to figure out the white backgrounds, if anyone knows how to bypass those please tell me

LemonInTheDark replied:
I hacked it, but it does work so eh

LemonInTheDark commented on code/datums/ruins.dm:
```diff
@@ -18,6 +18,7 @@
 	var/suffix = null
 
 	var/ruin_type = null
+	var/ignore = FALSE
```
debug var used to track maps I passed over, will remove when I'm all clear

Kylerace commented on code/modules/lighting/lighting_source.dm:
```diff
@@ -205,49 +216,82 @@
+	// This could be variable, it just isn't yet yaknow?
+	return max(multiplier * (1 - max(angle_delta - (angle / 2), 0) / 30), 0)
+
+/datum/light_source/proc/print_sheet()
```
what is a sheet?

LemonInTheDark replied:
I'll doc it but uh
Sheets are 2 dimensional lists that encode all the possible lighting values for a set of light vars (angle, dir, intensity, range)
We access them to pull out multipliers when passing out light values to corners.

LemonInTheDark commented on code/modules/lighting/lighting_source.dm:
```diff
@@ -165,7 +176,7 @@
 
 // Read out of our sources light sheet, a map of offsets -> the luminosity to use
-#define LUM_FALLOFF(C)  _sheet[C.x - _turf_x + _range_offset][C.y - _turf_y + _range_offset]
+#define LUM_FALLOFF(C) read_sheet(_sheet, C.x - _turf_x, C.y - _turf_y, _range_offset)// _sheet[C.x - _turf_x + _range_offset][C.y + _range_offset]
```
debug code, read_sheet is there so I can check in the debugger to make sure I'm not accessing the sheets wrong. will undo.

LemonInTheDark committed:
BIRDSHOT: Cargo and Engi

MMMiracles committed:
tram lighting rework

LemonInTheDark committed:
Merge pull request #5 from MMMiracles/tramlighting

tram lighting rework

LemonInTheDark committed:
Birdshot ai sat

LemonInTheDark committed:
Merge branch 'light-debug' of https://github.com/LemonInTheDark/tgstation into light-debug

LemonInTheDark committed:
BIRDSHOT: DONE! some debug fixes too

LemonInTheDark commented:
Alllright now it's just centcom, tram maint and *shudders* northstar

LemonInTheDark committed:
NORTH STAR: 1st floor (Arrivals, Departures? Cargo, Engi, spot of Med)

LemonInTheDark committed:
NORTHSTAR: 2nd floor. Medbay, part of sci, common space, library, parts of storage/ai stuff

LemonInTheDark committed:
NORTH STAR: Layer 3: Kitchen, bar, botany, some sci, dorms

LemonInTheDark committed:
NORTHSTAR: 4th layer: Sec, command, ai sat

LemonInTheDark committed:
NORTHSTAR: Touchups, maint, ALL THE STATIONS ARE DONE!!!

LemonInTheDark committed:
All of centcom. I've swapped out its fullbright for light emitters, and done the legwork to ensure that works properly. This + dimming/cleaning things out makes the space look a lot nicer I think

LemonInTheDark committed:
meta cleanup

LemonInTheDark committed:
meta cleanup

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into light-debug

LemonInTheDark committed:
reee

LemonInTheDark committed:
map and code cleanup. I am so tired

LemonInTheDark committed:
more cleanup, I will burn your crops

LemonInTheDark commented:
Very nearly done, just need to clean up some mapping errors

LemonInTheDark committed:
makes the corner stuff work, makes mapping helpers layer over everything by default, dme shicanary

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into light-debug

LemonInTheDark committed:
linter fix pack version 1

LemonInTheDark committed:
whoops

LemonInTheDark committed:
god damn it

LemonInTheDark committed:
map errors

LemonInTheDark committed:
should be good?

LemonInTheDark committed:
fuck off

LemonInTheDark committed:
final cleanup

san7890 commented on code/modules/cargo/centcom_podlauncher.dm:
```diff
@@ -34,6 +34,7 @@
 		/obj/effect/particle_effect/sparks,
 		/obj/effect/pod_landingzone,
 		/obj/effect/client_image_holder,
+		/obj/effect/light_emitter/podbay
```
```suggestion
		/obj/effect/light_emitter/podbay,
```

i am sorry uncle

san7890 commented on code/modules/lighting/lighting_atom.dm:
```diff
@@ -106,17 +120,38 @@
+		return
+	. = light_angle
+	light_angle = new_value
+	SEND_SIGNAL(src, COMSIG_ATOM_UPDATE_LIGHT_ANGLE, .)
```
```suggestion
	SEND_SIGNAL(src, COMSIG_ATOM_UPDATE_LIGHT_ANGLE, .)
	return .
```

dot operator tax

LemonInTheDark replied:
I KNOW MY RIGHTS, IT'S ONLY GOT ONE VALUE YOU CAN'T DO THIS TO ME

san7890 commented on code/modules/lighting/lighting_atom.dm:
```diff
@@ -106,17 +120,38 @@
+		return
+	. = light_dir
+	light_dir = new_value
+	SEND_SIGNAL(src, COMSIG_ATOM_UPDATE_LIGHT_DIR, .)
```
```suggestion
	SEND_SIGNAL(src, COMSIG_ATOM_UPDATE_LIGHT_DIR, .)
	return .
```

san7890 commented on code/modules/lighting/lighting_atom.dm:
```diff
@@ -126,10 +161,28 @@
 		return
 	. = light_flags
 	light_flags = new_value
 	SEND_SIGNAL(src, COMSIG_ATOM_UPDATE_LIGHT_FLAGS, .)
```
```suggestion
	SEND_SIGNAL(src, COMSIG_ATOM_UPDATE_LIGHT_FLAGS, .)
	return .
```

since we're here?

san7890 commented on code/modules/lighting/lighting_source.dm:
```diff
@@ -140,23 +151,23 @@
-	var/_applied_lum_b = lighting_source.applied_lum_b;                    \
-	var/_lum_r = lighting_source.lum_r;                                    \
-	var/_lum_g = lighting_source.lum_g;                                    \
+#define SETUP_CORNERS_CACHE(lighting_source)                                                               \
```
golly this is a wacky diff that's still mal-aligned, doesn't have to be fixed now but it'd look a heck of a lot better with just one space after each line rather than exacerbating the issue with more tabs. just FYIing for future improvement

LemonInTheDark replied:
the idea is to hide the \n part of it as much as we can, and have it line up nice.
it's how pjb did macros, and it's just carried over from then

san7890 replied:
yeah but it doesn't line up nice

san7890 commented on code/modules/lighting/lighting_source.dm:
```diff
@@ -205,50 +216,98 @@
+				print_column += round(row, 0.1)
+			output += print_column.Join(", ")
+		output += column_seperator
+	message_admins("\n[output.Join("\n")]")
```
message admins seems like it could get annoying in case someone wants to quickly check something wacky on prod during a test merge of a new map or something, would you mind passing in a user or something?

LemonInTheDark replied:
I mean sure

san7890 reviewed (changes requested):
simple and quick nits code-wise, everything seemed alright otherwise.

as far as maps are concerned, it is thoroughly too deep to do a pass on. stuff can be adjusted in post on a case by case basis, since if the design looks roughly the same as the examples you were showing me earlier in this project- i am satisfied.

LemonInTheDark committed:
san's review (I WILL HAVE MY REVENGE)

LemonInTheDark commented:
> as far as maps are concerned, it is thoroughly too deep to do a pass on. stuff can be adjusted in post on a case by case basis, since if the design looks roughly the same as the examples you were showing me earlier in this project- i am satisfied.

The images I posted in this pr are of what amount to "crummy cases" as examples of what I want to fix.
I'm not sure if that's what you're going off but I wanna avoid a false bill of sale.

LemonInTheDark commented:
icebox is a spurious (I hope to god) failure

san7890 commented:
![image](https://github.com/tgstation/tgstation/assets/34697715/f5fb3fde-92ea-4a93-ac00-b772f963e99a)

we'll find out if its not
